src/mongoc/mongoc-write-concern.c

```diff
@@ -98,7 +100,7 @@ bool
 mongoc_write_concern_get_fsync (const mongoc_write_concern_t *write_concern)
 {
    bson_return_val_if_fail(write_concern, false);
-   return write_concern->fsync_;
+   return (write_concern->fsync_ != MONGOC_WRITE_CONCERN_FSYNC_DEFAULT && !!write_concern->fsync_);
```

Comment:
Are all of these checks basically:

(x != -1 && !!x)

?

If so, let's just move to == 1